packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,260 @@
+        {props.children}
+      </DefaultSelectionTracker>
+    </RadioGroup>
+  );
+}
+
+function DefaultSelectionTracker(props: DefaultSelectionTrackProps) {
+  let state = useContext(RadioGroupStateContext);
+  let isRegistered = useRef(!(props.defaultValue == null && props.value == null));
+
+  // default select the first available item
+  let register = useCallback((value: string) => {
+    if (state && !isRegistered.current) {
+      isRegistered.current = true;
+      state.setSelectedValue(value);
+    }
+  }, []);
+
+  return (
+    <Provider
+      values={[
+        [InternalSegmentedControlContext, {register: register, prevRef: props.prevRef, currentSelectedRef: props.currentSelectedRef}]
+      ]}> 
+      {props.children}
+    </Provider>
+  );
+}
+
+function SegmentedControlItem(props: SegmentedControlItemProps, ref: FocusableRef<HTMLLabelElement>) {
+  let inputRef = useRef<HTMLInputElement>(null);
+  let domRef = useFocusableRef(ref, inputRef);
+  let divRef = useRef<HTMLDivElement>(null);
+  let {register, prevRef, currentSelectedRef} = useContext(InternalSegmentedControlContext);
+  let state = useContext(RadioGroupStateContext);
+  let isSelected = props.value === state?.selectedValue;
+  // do not apply animation if a user has the prefers-reduced-motion setting
+  let isReduced = false;
+  if (window?.matchMedia) {
+    isReduced = window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+  }
+
+  useLayoutEffect(() => {
+    register?.(props.value);
+  }, []);
+
+  useLayoutEffect(() => {
+    if (isSelected && prevRef?.current && currentSelectedRef?.current && !isReduced) {
+      let currentItem = currentSelectedRef?.current.getBoundingClientRect();
+
+      let deltaX = prevRef?.current.left - currentItem?.left;
+
+      currentSelectedRef.current.animate(
+        [
+          {transform: `translateX(${deltaX}px)`, width: `${prevRef?.current.width}px`},
+          {transform: 'translateX(0px)', width: `${currentItem.width}px`}
+        ],
+        {
+          duration: 200,
+          easing: 'ease-out'
+        }
+      );
+
+      prevRef.current = null;
+    }
+  }, [isSelected]);
+
+  return (
+    <Radio 
+      {...props} 
+      ref={domRef} 
+      inputRef={inputRef}
+      style={props.UNSAFE_style}
+      className={renderProps => (props.UNSAFE_className || '') + controlItem({...renderProps}, props.styles)} >
+      {({isSelected, isFocusVisible, isPressed, isDisabled}) => (
+        <>
+          {isSelected && <div className={slider({isFocusVisible, isDisabled})} ref={currentSelectedRef} />}
+          <Provider 
+            values={[
+              [IconContext, {
+                render: centerBaseline({slot: 'icon', styles: style({order: 0, flexShrink: 0})})
+              }],
+              [RACTextContext, {slots: {[DEFAULT_SLOT]: {}}}],
+              [TextContext, {styles: style({order: 1, truncate: true})}]
```

Comment:
oh hm, I guess this was supposed to apply the truncation, not sure why it isn't taking